x/capability/keeper/keeper.go

```diff
@@ -113,22 +105,38 @@ func (k *Keeper) InitializeAndSeal(ctx sdk.Context) {
 		panic(fmt.Sprintf("invalid memory store type; got %s, expected: %s", memStoreType, sdk.StoreTypeMemory))
 	}
 
-	prefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixIndexCapability)
-	iterator := sdk.KVStorePrefixIterator(prefixStore, nil)
+	k.sealed = true
+}
```

Comment:
so `InitializeAndSeal` no longer initializes any capabilities, it just seals the keeper and ensures the mem store is a mem store. Should the godoc be updated?

Reply:
Correct and yes, I break this API on the breaking-changes pr

Reply:
This function doesn't do a full initialization - it doesn't call `InitMemStore`. Maybe let's make a note, that this function should be called after `InitMemStore`. So we should update the comments of this function and `InitMemStore` function. And in 0.43 we can maybe rework it to make it more clear - by renaming this function to `k.Seal(ctx)`

BTW: we don't need to check the store type - this should be done in `InitMemStore`.

Reply:
I see that you are sealing in the `InitMemStore`. So we should update the comments of this function and `InitMemStore`. And in `0.43` we can maybe rework it to make it more clear.

Reply:
Also, this is a breaking change of the state machine.

Reply:
No this shouldn't be breaking the state machine. `InitializeCapability` only writes to the memstores and in-memory go map.

The memory store **does not** get committed along with the rest of the app state. It is in-memory local storage for each node. So changing things inside of it does not break the state machine.

https://github.com/cosmos/cosmos-sdk/blob/master/store/mem/store.go

Reply:
> The memory store does not

I know. Breaking state machine doesn't necessary mean to break the written state. To illustrate this issue:
1. Node A will run app with 0.42.8 
2. Node B will run same app (same app manager config) but using 0.42.9

Node A will have all capabilities initialized, Node B will not have.

Reply:
No this PR just changes where/when the capabilities are initialized.

In 0.42.8, the capabilities are initialized in app start `NewApp`, and then reinitialized on the first transaction that calls `GetCapability`.

In 0.42.9, the capabilities are initialized at the first abci method after startup (either `InitChain` or `BeginBlock`).

In either case, the capabilities are initialized **before** they are requested in a transaction, which is what is necessary for keeping nodes in sync.

There is a bug in 0.42.7 and 0.42.8, so there are situations where its state becomes out of sync with a node from 0.42.6.

However, comparing a node on 0.42.6 to a node on 0.42.9. They may initialize the in-memory capabilities in different places; however both will be fully initialized by the time the capability module gets used and thus will authenticate and reject the same transactions and be perfectly in sync.

Reply:
The behavior of `InitializeCapability` is different - so an app will need to make more changes in the code than bumping a version. IMHO this is a breaking change - we signal to the app that bumping a version is not enough.

Reply:
Yes just bumping the version will not be enough. You will also have to make changes to `app.go`. But it should be clear that you can run gaia with `0.42.6` and `0.42.9` on the same network and there will be no state divergence